synapse: update to 1.37.1

AluminumTank opened the pull request:
<!-- Mark items with [x] where applicable -->
New dependency `python3-ijson` needed.
#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

<!--
If GitHub CI cannot be used to validate the build result (for example, if the
build is likely to take several hours), make sure to
[skip CI](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration).
When skipping CI, uncomment and fill out the following section.
Note: for builds that are likely to complete in less than 2 hours, it is not
acceptable to skip CI.
-->
<!-- 
#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [ ] I built this PR locally for my native architecture, (ARCH-LIBC)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl
-->

FollieHiyuki commented:
You can add `yajl` to `depends` of the new package, and also `python3-cffi` to `checkdepends` to pass tests.

AluminumTank commented:
Thanks for the tip. I added those as well as the `python3-devel` and `yajl-devel` files to `makedepends` so more of the backends get built. The tests are failing on a different issue now: looking for the `yajl` backend which depends on YAJL 1.x.

AluminumTank commented:
Confirmed that all backends are working other than `yajl` backend. Since this backend depends on YAJL 1.x, it shouldn't be supported since the `yajl` package in `void-packages` is 2.x.x. I'm no expert here, but it seems like we need to skip just that one test...

AluminumTank commented:
Updated to latest release and added log service so I can stop manually adding it on my system. Still need help/clarification on whether we should do ci-skip or patch the test to not try using the YAJL 1.x backend.

ericonr commented on srcpkgs/synapse/files/synapse/log/run:
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+exec logger -p daemon.notice -t synapse
```
Please do like other services and use a symlink to `/usr/bin/vlogger`

ericonr commented:
It should be possible to skip the test with a custom `do_check` or `make_check_args`. If not, patching it out is valid.

ericonr commented on srcpkgs/python3-ijson/patches/remove-yajl-backend.patch:
```diff
@@ -0,0 +1,62 @@
+--- ijson/backends/yajl.py.orig	2021-03-02 05:04:56.000000000 -0500
```
Could you instead simply remove the file in `post_patch`?

`rm ijson/backends/yajl.py`

Kinda weird that we build the yajl stuff only to completely remove it :[

AluminumTank replied:
`yajl.py` is the backend for yajl 1.x. There are 3 other backends left intact to interact with the version of yajl already packaged (2.x). I was removing this because simply having the patch to remove the backend from `test/test_base.py` wasn't enough to pass the check phase.

ericonr commented on srcpkgs/python3-ijson/template:
```diff
@@ -0,0 +1,23 @@
+checksum=1d1003ae3c6115ec9b587d29dd136860a81a23c7626b682e2b5b12c9fd30e4ea
+
+post_patch() {
+	rm ijson/backends/yajl.py
```
Please leave a comment about https://github.com/void-linux/void-packages/pull/31263#discussion_r653196518 on top of it.

AluminumTank commented:
Alright, I found the underlying problem that made me need to patch out the `yajl` (1.x) backend: the `python3-module.sh` build-style script defaults to running `python3 setup.py test` if `pytest` isn't included in `checkdepends`. Adding this makes the patching out method unnecessary, so I removed. Package should be ready for review now.

ericonr commented:
Nice! Can you split the commits? Should be one for new package and one for the update. I can merge it manually later, if not.

AluminumTank commented:
Yep, done.
